app/services/courses/fetch.rb

```diff
@@ -0,0 +1,38 @@
+module Courses
+  class Fetch
+    class << self
+      def by_code(provider_code:, course_code:, cycle_year: Settings.current_recruitment_cycle_year)
+        Course
+          .includes(:subjects)
+          .includes(:sites)
+          .includes(provider: [:sites])
+          .includes(:accrediting_provider)
+          .where(recruitment_cycle_year: cycle_year)
+          .where(provider_code: provider_code)
+          .find(course_code)
+          .first
+      end
+
+      def by_accrediting_provider(provider)
+        # rubocop:disable Style/MultilineBlockChain
+        # rubocop:disable Style/HashTransformValues
+        provider
+          .courses
+          .group_by do |course|
+            # HOTFIX: A courses API response no included hash seems to cause issues with the
```

Comment:
I don't imagine we'll see this on New Publish side, so can potentially simplify these few lines?